ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -113,9 +113,14 @@ public boolean isLeaderReady() {
       return isLeader() && getRole().isLeaderReady();
     }
 
+    @Override
+    public RaftPeer getCurrentLeader() {
+      return getRaftConf().getPeer(getState().getLeaderId());
+    }
+
     @Override
     public LifeCycle.State getLifeCycleState() {
-      return lifeCycle.getCurrentState();
+      return getLifeCycle().getCurrentState();
```

Comment:
Please revert this and the following unrelated changes.

Reply:
Fixed